purchase_work_acceptance/models/work_acceptance.py

```diff
@@ -0,0 +1,166 @@
+# Copyright 2019 Ecosoft Co., Ltd. (http://ecosoft.co.th)
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+from odoo import api, fields, models
+
+
+class WorkAcceptance(models.Model):
+    _name = 'work.acceptance'
+    _description = 'Work Acceptance'
+    _order = 'id desc'
+
+    @api.depends('wa_line_ids.price_total')
+    def _compute_amount_all(self):
+        for lines in self:
+            amount_untaxed = amount_tax = 0.0
+            for line in lines.wa_line_ids:
+                amount_untaxed += line.price_subtotal
+                amount_tax += line.price_tax
+            lines.update({
+                'amount_untaxed': lines.currency_id.round(amount_untaxed),
+                'amount_tax': lines.currency_id.round(amount_tax),
+                'amount_total': amount_untaxed + amount_tax,
+            })
+
+    name = fields.Char(
+        required=True,
+        index=True,
+        copy=False,
+        default='New'
+    )
+    origin = fields.Char(
+        string='Source Document',
+        copy=False,
+        help="Reference of the document that generated this work acceptance."
+    )
+    date_due = fields.Datetime(
+        string='Due Date',
+        required=True,
+        index=True,
```

Comment:
Normally, we will need index for the joining field, ie.., Many2one from other, which expect to have SQL Join
For normal filed, mostly we don't need index.